pkg/metrics/metrics.go

```diff
@@ -0,0 +1,123 @@
+package metrics
+
+import (
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+const (
+	// Registry-namespaces of operator sources.
+	REDHAT_OPERATOR_NS    = "redhat-operators"
+	COMMUNITY_OPERATOR_NS = "community-operators"
+	CERTIFIED_OPERATOR_NS = "certified-operators"
+
+	// Default values which are to be used as labels
+	// for prometheus metrics.
+	REGISTRY_NAMESPACE = "registry_ns"
+	OPSRC_NAME         = "opsrc_name"
+
+	// Path, Port and Host where custom metrics would be exposed.
+	MetricsPath = "/metrics"
+	MetricsPort = ":8383"
+	MetricsHost = "0.0.0.0"
+)
+
+// Metric variables that report error based on the operator-source
+// which fails to reconcile.
+var (
+	redhatErrorCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "redhat_operator_reconcile_failure_count",
+			Help: "Monotonic count of times redhat-opsrc was unable to reconcile",
+		},
+		[]string{REGISTRY_NAMESPACE, OPSRC_NAME},
+	)
+	communityErrorCount = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
```

Comment:
Are we able to expose a string as an error? Knowing how many times an operator failed is not helpful, knowing why it failed is.